Lib/test/test_threading.py

```diff
@@ -104,6 +104,11 @@ def test_various_ops(self):
             self.assertRegex(repr(t), r'^<TestThread\(.*, initial\)>$')
             t.start()
 
+        if hasattr(_thread, 'get_native_id'):
```

Comment:
Can you test threading.get_native_id() instead to avoid relying on private module?

Reply:
@vstinner The goal here is to determine whether the function is defined in C. `threading.get_native_id()` is always available in the Python module but will return `None` if the capability is not supported by the OS. 

If you think `threading.get_native_id()` shouldn't even exist on the Python side if the functionality isn't available, can you suggest a mechanism to remove the Python function only on unsupported platforms, or alternatively, a mechanism to avoid defining it altogether (on those unsupported platforms only)?